src/main/java/com/sysdig/jenkins/plugins/sysdig/scanner/InlineScannerRemoteExecutor.java

```diff
@@ -108,7 +108,8 @@ public String call() throws InterruptedException, AbortException {
     addProxyVars(envVars, containerEnvVars, logger);
 
     List<String> bindMounts = new ArrayList<>();
-    bindMounts.add("/var/run/docker.sock:/var/run/docker.sock");
+    String daemonSocket = envVars.get("DOCKER_HOST", "/var/run/docker.sock");
+    bindMounts.add(daemonSocket + ":/var/run/docker.sock");
```

Comment:
Should we check if this is a valid path? I guess this will work only when using Linux pipes for the docker sock, but I am afraid that the docker client might fail i.e. if DOCKER_HOST=tcp://192.168.1.56:5678 (it will try to bind mount an invalid path).

In the later case, when DOCKER_HOST is not a valid linux pipe path, we can just ignore the bind mount, and instead set DOCKER_HOST environment variable in the container so it connects to the right address?

Reply:
Also as discussed internally, DOCKER_HOST is also set internally in the container via DockerClientRunner class, so the docker client inside the container will search in the specified path instead of default /var/run/docker.sock when DOCKER_HOST is set. My proposal (without a full refactor of envvar handling inside DockerClientRunner) would be:
- Only add the bindMount when DOCKER_HOST is a valid path (otherwise it is tcp://... or similar)
- Don't handle DOCKER_HOST environment variable (from `currentEnv`) inside DockerClientRunner class. Instead, have an explicit parameter for the DockerHost and set the value when creating the runner in InlineScannerRemoteExecutor (see https://github.com/jenkinsci/sysdig-secure-plugin/blob/dacb8dbeeba257e18933165faba408219e324481/src/main/java/com/sysdig/jenkins/plugins/sysdig/scanner/InlineScannerRemoteExecutor.java#L87) so:
  - if DOCKER_HOST is not defined, use defaults (/var/run/docker.sock:/var/run/docker.sock) for bind mount and don't set DOCKER_HOST in the container
  - if DOCKER_HOST is defined, and it is a valid path, change the bind mount but mount in the default location, and don't set DOCKER_HOST inside the container (so bind mount is /custom/path/docker.sock:/var/run/docker.sock)
  - if DOCKER_HOST is defined and an invalid path, don't do the bind mount, and just set the value of DOCKER_HOST inside the container (i.e. DOCKER_HOST=tcp://x.x.x.x)
  -